pandas/core/groupby/grouper.py

```diff
@@ -520,21 +520,21 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     any_arraylike = any(isinstance(g, (list, tuple, Series, Index, np.ndarray))
                         for g in keys)
 
-    try:
-        if isinstance(obj, DataFrame):
-            all_in_columns_index = all(g in obj.columns or g in obj.index.names
-                                       for g in keys)
-        elif isinstance(obj, Series):
-            all_in_columns_index = all(g in obj.index.names for g in keys)
-        else:
+    if (not any_callable and not any_arraylike and not any_groupers and
+            match_axis_length and level is None):
+        try:
+            if isinstance(obj, DataFrame):
+                all_in_columns_index = all(g in obj.columns or g in
+                                           obj.index.names for g in keys)
+            elif isinstance(obj, Series):
+                all_in_columns_index = all(g in obj.index.names for g in keys)
+            else:
+                all_in_columns_index = False
+        except Exception:
             all_in_columns_index = False
```

Comment:
what test actually hits this exception?

Reply:
None of them, that was there before the PR that caused this regression.

Also, the
```
else:
    all_in_columns_index = False
````
can probably be removed as no tests hit that now either.  I mentioned that in my last PR, but didn't get any feedback/confirmation that could be removed.

Reply:
Is it possible to add tests that cover those lines of code?

Reply:
Just by using public functions?  I don't think it's reachable.  Probably not even reachable using `_get_grouper` directly.  I'll see what I can do.

Reply:
if you can remove code and nothing breaks then go for it